lib/editor/selectors/timetable.js

```diff
@@ -60,29 +61,39 @@ export const getTimetableColumns = createSelector(
     ]
     if (pattern && pattern.patternStops) {
       if (!pattern.useFrequency) {
-        pattern.patternStops.map((ps, index) => {
-          const stop = stops ? stops.find(st => st.stop_id === ps.stopId) : null
-          const stopName = stop ? stop.stop_name : ps.stopId
+        const patternHalts = mergePatternHalts(pattern.patternStops, pattern.patternLocations)
+        patternHalts.map((ps, index) => {
+          const isLocation = ps.hasOwnProperty('locationId')
+          let stop = null
+          // $FlowFixMe flow doesn't understand hasOwnProperty
+          if (stops && !isLocation) stop = stops.find(st => st.stop_id === ps.stopId)
+          // $FlowFixMe flow doesn't understand hasOwnProperty
+          if (locations && isLocation) stop = locations.find(l => l.location_id === ps.locationId)
+
+          // $FlowFixMe FLEX TODO: improve this to avoid double ternary
+          const stopName = stop ? stop.stop_name : ps.stopId ? ps.stopId : ps.locationId
           const stopId = stop && stop.stop_code
             ? stop.stop_code
             : stop && stop.stop_id
               ? stop.stop_id
-              : ''
+              : stop && stop.location_id ? stop.location_id : ''
```

Comment:
Some of these ternary expressions can be simplified.

Reply:
These haven't been touched in 4 years which is why I'm weary of potentially introducing more unexpected regressiosn